homeassistant/setup.py

```diff
@@ -19,6 +19,9 @@
 DATA_DEPS_REQS = "deps_reqs_processed"
 
 SLOW_SETUP_WARNING = 10
+# Since a pip install can run, we wait
+# 30 minutes to timeout
+SLOW_SETUP_MAX_WAIT = 1800
```

Comment:
The default timeout of HA to cancel a startup is 10min. Most user will never see that

Reply:
I don't think that is actually happening because I see this in the log
```
2020-05-24 17:21:02 ERROR (MainThread) [homeassistant.setup] Setup of component nut is taking longer than 1800 seconds. Startup will proceed without waiting any longer.
2020-05-24 17:21:02 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform nut.sensor: Unable to set up component.
```

Reply:
Pascal is talking about installations that run under the Supervisor. The Supervisor is doing a health check to see if the API is coming online, if so, it marks the upgrade as successful.

Reply:
However, if the frontend is available via #36093, this wouldn't be an issue.